fix: Ensure visual indicating for focus loss.

BenHenning opened the pull request:
Fixes #92

If the workspace loses focus, any existing block selection or cursor visuals should be hidden to represent that the focus has been lost. Instead, the passive indicator should be shown so that the user has context on where they are returning when navigating back to the workspace.

This changes the hide behavior for the line cursor to also clear any block selection that may be present (since the cursor is using selection as a means of visually highlighting blocks today).

There was an issue restoring selection due to the passive focus indicator interfering with block selection (since it was using selection). This was updated to use non-selection CSS to keep the indicator without affecting the selection state. This pulls in the changes from #236.

See video for the new state:

[Screen recording 2025-03-11 4.15.41 PM.webm](https://github.com/user-attachments/assets/2b82f8a2-4e16-471f-b579-e0908d8dc6df)

BenHenning committed:
fix: Ensure visual indicating for focus loss.

If the workspace loses focus, any existing block selection or cursor
visuals should be hidden to represent that the focus has been lost.

BenHenning commented on src/navigation.ts:
```diff
@@ -1139,7 +1139,7 @@ export class Navigation {
     // in the base Marker class and this can therefore be null.
     if (this.markedNode) {
-      this.passiveFocusIndicator.show(this.markedNode);
+      // this.passiveFocusIndicator.show(this.markedNode);
```
TODO: Get resolution from Rachel and/or Christopher before finalizing this solution and moving PR to review.

cpcallen replied:
I think we want to continue to display a passive focus indicator here.  I agree that is is suboptimal that it is not distinguishable from the active focus indicator—but I think the fix to that is to change how it looks, not to not display it at all.  @rachel-fenichel ?

BenHenning replied:
This has been fixed in the latest, PTAL.

rachel-fenichel committed:
feat: use CSS for passive focus on blocks and next connections

BenHenning committed:
Merge remote-tracking branch 'rachel-fenichel/passive_focus_css' into indicate-when-workspace-focus-is-lost

BenHenning committed:
Attempt to use passive focus indicator via CSS.

BenHenning committed:
Add a workaround to fix passive indicator hiding.

This was a workable solution found by Rachel with the explanation
captured in a new line comment.

BenHenning committed:
Merge branch 'main' into indicate-when-workspace-focus-is-lost

Conflicts:
	src/line_cursor.ts

BenHenning committed:
Use a better color for passive indicators.

This color came at Rachel's suggestion.
